applications/RomApplication/custom_strategies/rom_builder_and_solver.h

```diff
@@ -438,88 +455,179 @@ class ROMBuilderAndSolver : public BuilderAndSolver<TSparseSpace, TDenseSpace, T
 
         // Assemble all entities
         const auto assembling_timer = BuiltinTimer();
-        #pragma omp parallel firstprivate(nelements, nconditions, LHS_Contribution, RHS_Contribution, EquationId, el_begin, cond_begin)
-        {
-            Matrix PhiElemental;
-            Matrix tempA = ZeroMatrix(mNumberOfRomModes,mNumberOfRomModes);
-            Vector tempb = ZeroVector(mNumberOfRomModes);
-            Matrix aux;
-
-            #pragma omp for nowait
-            for (int k = 0; k < static_cast<int>(nelements); k++) {
-                auto it_el = el_begin + k;
+        if (mSolveWithQR){// QR POD: This is used to build Petrov-Galerkin POD.

```

Comment:
define a functin for QRAssembly and StandardROMAssembly )or the names you find proper. it will make the code much clearer